File: Exercism/javascript/beer-song/beer-song.js
'use strict'

const beer = (i) => `${i || 'no more'} bottle${i !== 1 ? 's' : ''} of beer`
const wall = (i) => `${beer(i)} on the wall`

export class BeerSong {
  static verse (i) {
    return `${wall(i).toUpperCase()[0] + wall(i).slice(1)}, ${beer(i)}.\n` +
      (i === 0
        ? `Go to the store and buy some more, ${wall(99)}.\n`
        : `Take ${i !== 1 ? 'one' : 'it'} down and pass it around, ${wall(i - 1)}.\n`)
  }

  static sing (from = 99, to = 0) {
    return this.verse(from) +
      (from === to ? '' : '\n' + this.sing(from - 1, to))
  }
}

File: Exercism/python/beer-song/beer_song.py
def recite(start: int, take: int = 1) -> list:
    """List the famous beer song"""
    song = list()
    for i in range(start, start-take, -1):
        if i == 0:
            song.append("No more bottles of beer on the wall, "
                        "no more bottles of beer.")
            song.append("Go to the store and buy some more, "
                        "99 bottles of beer on the wall.")
            break

        song.append(f"{i} bottle{'s' if i > 1 else ''} of beer on the wall, "
                    f"{i} bottle{'s' if i > 1 else ''} of beer.")
        song.append((f"Take one down and pass it around, "
                     f"{i - 1} bottle{'s' if i - 1 > 1 else ''} "
                     f"of beer on the wall.")
                    if i > 1 else
                    (f"Take it down and pass it around, "
                     f"no more bottles of beer on the wall."))
        if i > start - take + 1:
            song.append("")

    return song
